recon/src/libp2p/handler.rs

```diff
@@ -200,12 +201,30 @@ where
                         }
                         Err(e) => {
                             return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
-                                FromHandler::Failed(e),
+                                FromHandler::TransientError(e),
                             ))
                         }
                     }
                 }
             }
+            State::Inbound(stream) => {
```

Comment:
Why are Inbound and Outbound different?

Reply:
My thinking was we want to mark inbound as errors still (potentially, since it might have been the peers fault making it failed makes sense). But outbound means we encountered a problem on our node, so we shouldn't mark the peer as failed.